lib/widgets/totals_table.dart

```diff
@@ -68,32 +84,56 @@ class TotalsTable extends StatelessWidget {
     for (var item in [...fodderItems, ...concentrateItems]) {
       for (var col in columns) {
         String key = col['key'] as String;
-        totals[key] = (totals[key] ?? 0) + (item[key] as num);
+        totals[key] = (totals[key] ?? 0) + (item[key] ?? 0);
       }
     }
 
+    double totalConcentrateDM = 0;
+
+    // dmIntake totals for concentrate items
+    for (var item in concentrateItems) {
+      totalConcentrateDM += item['dmIntake'] ?? 0;
+    }
+    totals['concentrateIntake'] =
+        totalConcentrateDM / totals['dmIntake']! * 100;
+
     return totals;
   }
 
-  Widget _buildComparisonIcon(
-      double total, double requirement, bool isPercentage) {
-    const double percentageThreshold = 2;
-    const double absoluteThreshold = 0.05;
+  Widget _buildComparisonIcon(String key, double total, bool isLactating) {
+    Map<String, Map<String, double>> limits = {
+      // TODO: update concentrateIntake range
+      'concentrateIntake': {'min': 0, 'max': 100},
+      'caIntake': {
+        'min': isLactating ? 0.7 : 0.35,
+        'max': isLactating ? 1.0 : 0.55,
+      },
+      'pIntake': {
+        'min': isLactating ? 0.35 : 0.25,
+        'max': isLactating ? 0.45 : 0.40,
+      },
+      'ndfIntake': {'min': 28, 'max': 35},
```

Comment:
There is a bug here. If a cow eats 10kg of Brewers grain (and only brewers grain), then the cow has 8kg of dry matter, and 3.2kg of NDF. This means that they have 40% NDF, but in your app there seems to be an answer of 3.2%?

Reply:
ah, so NDF should be 40% of DM?

Reply:
actually CP, NDF, Ca, P should all be %DM in the totals table. I understand now; I'll fix it. I was just summing up the values - the values themselves should be kg/d, then totals should be % DM.